actors/builtin/market/market_actor.go

```diff
@@ -82,7 +81,6 @@ func (a Actor) WithdrawBalance(rt Runtime, params *WithdrawBalanceParams) *adt.E
 	rt.State().Transaction(&st, func() interface{} {
 		// Before any operations that check the balance tables for funds, execute all deferred
 		// deal state updates.
-		amountSlashedTotal = big.Add(amountSlashedTotal, st.updatePendingDealStatesForParty(rt, nominal))
```

Comment:
What if someone calls this method after its deal collateral is marked slashed but not yet processed by Cron? and Cron is the last message in a tipset?

Reply:
Good thought. Penalties are probably worth processing immediately, and we know exactly what the deal IDs are in that case anyway.

Reply:
this is no longer an issue because we are only withdrawing available funds that is not in the locked table. We should also test this.